src/collectors/community/ensap/ensap.ts

```diff
@@ -0,0 +1,139 @@
+                autoDownloadModelName: 'large-v3-turbo', // (optional) auto download a model if model is not present
+                removeWavFileAfterTranscription: true, // (optional) remove wav file once transcribed
+                whisperOptions: {
+                    outputInJson: true,
+                    language: 'fr'
+                },
+                logger: {
+                    debug: (...args: any[]) => {},
+                    error: (...args: any[]) => {console.error(args)},
+                    log: (...args: any[]) => {console.log(args)}
+                }
+            });
+
+            const whisper_result = fs.readFileSync(audio_path + '.json', 'utf8');
+            const whisper_data = JSON.parse(whisper_result);
+            fs.unlinkSync(audio_path + '.json');
+
+            const captcha_result = whisper_data.transcription.map(item => item.text).join('');
+
+            const digits = captcha_result.match(/\d/g);
+
+            if (digits == null || digits.length < 8) {
+                count++;
+                continue;
+            }
+
+            const firstEightDigits = digits.join('').substring(0, 8);
+
+            // Input identifier, password and captcha
+            await driver.inputText(EnsapSelectors.FIELD_IDENTIFIER, params.id, { delay: 500 });
+            await driver.inputText(EnsapSelectors.FIELD_PASSWORD, params.password);
+            await driver.inputText(EnsapSelectors.FIELD_CAPTCHA, firstEightDigits);
+
+            // Click on submit button
+            await driver.leftClick(EnsapSelectors.BUTTON_SUBMIT);
+
+            const login_alert = await driver.getElement(EnsapSelectors.CONTAINER_LOGIN_ALERT, { raiseException: false, timeout: 2000 });
+            if (login_alert) {
+                return await login_alert.textContent("i18n.collectors.all.password.error");
+            }
+
+            is_logged_in = (driver.url() === "https://ensap.gouv.fr/web/accueilconnecte");
+            count++;
+        }
+
+        // Check if login error exists
+        if (!is_logged_in) {
+            const login_validation = await driver.getElement(EnsapSelectors.CONTAINER_LOGIN_VALIDATION, { raiseException: false, timeout: 2000 });
+            if (login_validation) {
+                return await login_validation.textContent("i18n.class.state.1_error");
+            }
+            return "i18n.class.state.1_error";
+        }
+    }
+
+    async collect(driver: Driver, params: any): Promise<Invoice[]> {
+        // Get documents
+        const {requestBody, responseBody} = await driver.goto(EnsapCollector.CONFIG.entryUrl, "prive/remunerationpaie/v1");
+
+        // Build return array
+        return await Promise.all(responseBody.map(async document => {
+            const timestamp = new Date(document.dateDocument).getTime();
+            const link = "https://ensap.gouv.fr/prive/telechargerremunerationpaie/v1?documentUuid=" + document.documentUuid;
```

Comment:
Use `${document.documentUuid}` instead of + operator

Reply:
Done